OMEdit/OMEdit/OMEditGUI/Modeling/ModelWidgetContainer.cpp

```diff
@@ -1503,8 +1500,28 @@ void GraphicsView::addItem(QGraphicsItem *pGraphicsItem)
  */
 void GraphicsView::removeItem(QGraphicsItem *pGraphicsItem)
 {
-  if (scene()->items().contains(pGraphicsItem)) {
+  if (mAllItems.contains(pGraphicsItem)) {
```

Comment:
I think we also want to postpone the removal here when `mToBeCleared` flag is set.

Reply:
Do you mean calling `emitDeleted`? Because `mAllItems` is cleared anyway just before scene is cleared.

Reply:
No. I mean the if should be like this,

`if (!mToBeCleared && mAllItems.contains(pGraphicsItem)) {`